api/C/include/csi_ll_pmp.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Low-Level PMP API
+ *
+ * This module provides functionality for PMP configuration.
+ *
+ * Copyright (c) RISC-V International 2023. Creative Commons License. Auto-
+ * generated file: DO NOT EDIT
+ */
+
+#ifndef CSI_LL_PMP_H
+#define CSI_LL_PMP_H
+
+#include "csi_types.h"
+
+/*
+ * Get PMP range number for the hard id. If the hart id is invalid, such as out of
+ * support range, it should return an error
+ *
+ * @param hart_id: The hart id used to get the pmp entry number
+ * @return : The number of PMP entry; -1 will be returned as and error code if the
+ * input is invalid.
+ */
+int csi_pmp_get_entry_num(unsigned int hart_id);
+
+/*
+ * Set PMP entry. This function only can be called in M-mode. If it is called in
+ * other mode, it may cause carsh. When this function is called with success, it
+ * will not be effective untill  the core exits M-mode. Also, an error should be
+ * returned if the output parameters is invalid, such as idx out of range, address
+ * or size is not supported or over overlapped etc.
+ *
+ * @param idx: The PMP entry index
+ * @param base_addr: The base address of this range need to be configured
```

Comment:
If the address matching is TOR, what does it do with base address?  Is it ignored, or does it modify the address of the previous entry?

Reply:
Yes, my idea is that it should be ignored. And the address of previous entry is considered as base address. Do you have good idea about it?

Reply:
I think this is probably OK if the behaviour is explained in the comments.